js/src/components/lineage/LineageView.tsx

```diff
@@ -370,11 +374,7 @@ export function PrivateLineageView(
         filteredNodeIds = viewOptions.node_ids;
       } else {
         const packageName = lineageGraph.manifestMetadata.current?.project_name;
-        const viewMode = viewOptions.view_mode
-          ? viewOptions.view_mode
-          : isModelsChanged
-            ? "changed_models"
-            : "all";
+        const viewMode = (viewOptions.view_mode ?? isModelsChanged) ? "changed_models" : "all";
```

Comment:
The statement could produce a wrong result.
The `viewMode` can be "changed_models" when `viewOptions.view_mode` is "all".

Reply:
Fixed